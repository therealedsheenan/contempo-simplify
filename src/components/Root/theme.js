// @flow

import { primary, secondary } from '../../styles/color';

const theme = {
  primary,
  secondary
};

export default theme;
