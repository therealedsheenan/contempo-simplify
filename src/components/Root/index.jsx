// @flow

import React from 'react';

// $FlowFixMe
import { ThemeProvider } from 'styled-components';

import theme from './theme';

// reset styles
import '../../styles/base';

const Root = (props: { children: Object }) => (
  <ThemeProvider theme={theme}>
    <main>
      {props.children}
    </main>
  </ThemeProvider>
);

export default Root;
