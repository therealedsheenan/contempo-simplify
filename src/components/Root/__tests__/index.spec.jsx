import React from 'react';
import renderer from 'react-test-renderer';

import Root from '../index';

const mockComponent = () => <div>My component.</div>;

test('Home existence', () => {
  const component = renderer
    .create(
      <Root>
        <mockComponent />
      </Root>
    )
    .toJSON();

  // correct home component
  expect(component).toMatchSnapshot();
});
