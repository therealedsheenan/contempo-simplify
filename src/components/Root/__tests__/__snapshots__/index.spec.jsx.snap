// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home existence 1`] = `
<main>
  <mockComponent />
</main>
`;
